### PR TITLE
Modernise role

### PR DESCRIPTION
**Pull request by sjpb**
- Replaces deprecated `os_client_config` with `openstack` SDK package; role now supports using application credentials.
- Adds support for defining multiple shares/mounts in a single role invocation.
- Updates default Ceph version to `pacific`, with corresponding changes to repo/key URLs.
- Runs lookup action on localhost by default.
- Provides separate `install.yml` task file to support use in image build.
- More configuration for mount state, mount options etc.

**NB**: This PR makes breaking changes to variables:
- `os_manila_mount_action`: Removed. Use parameter `task_from` on module `import_role` instead.
- `os_manila_mount_share_{name,user,protocol}`: Replaced by `os_manila_mount_shares`.
- `os_manila_mount_{path,user,group}`: Replaced by `os_manila_mount_shares`.
- `os_manila_mount_os_config_name`: Removed, no longer required.
- `os_manila_mount_{host,export,access_key}`: Replaced by `os_manila_mount_share_info`.
- `os_manila_mount_pkgs_install`: Removed, don't run `install.yml` task if packages not required.
- `os_manila_mount_ceph_repo_base`: Replaced by `os_manila_mount_ceph_release_repo`.
- `os_manila_mount_fuse`: Removed, only kernel driver is now supported.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,39 +0,0 @@
...
-  # Obtain custom yamllint rules for Ansible
-  - curl -O https://raw.githubusercontent.com/ansible/galaxy/devel/galaxy/importer/linters/yamllint.yaml
-
-script:
-  # Basic role syntax check
-  - ansible-playbook tests/test.yml -i tests/inventory --syntax-check
-  # Ansible lint
-  - ansible-lint tests/test.yml
-  # YAML lint
-  - yamllint -c yamllint.yaml .
```
Comment (JohnGarbutt) [inline .travis.yml:36]: I would like to keep the CI though, is it easy to add the github actions in here... I guess we have a standard way to do this now?

File: README.md [hunk 2]
```diff
@@ -1,183 +1,111 @@
...
-* `os_manila_mount_share_name`: Name in Manila for the share.  This name is also
-  used in the output of `manila list`.
-* `os_manila_mount_share_user`: Cluster user name for mount access check,
-  and subsequent mount operation.
-* `os_manila_mount_share_protocol`: Filesystem type to look up.  Currently 
-  supported options are `CEPHFS`.
-* `os_manila_mount_os_config_name`: Cloud config name (if applicable),
-  as defined in `/etc/openstack/clouds.yaml`
+- `python-openstackclient`
+- `python-manilaclient`
```
Comment (JohnGarbutt) [inline README.md:16]: I think we should also add "openstacksdk" to the list, to make the module work?

File: library/os_manila_share.py [hunk 3]
```diff
@@ -1,122 +1,65 @@
...
+    
+    # the share object doesn't actually expose lots of stuff from the API, including the exports
+    share_details = conn.share.get(f"/shares/{share.id}").json()['share']
+    # module.fail_json(share_details) # DEBUG
+    protocol = share_details['share_proto'].upper()
+    if protocol != 'CEPHFS':
+        module.fail_json("Share named %r is protocol %r, can only handle CEPHFS" % (share_name, protocol))
+    
+    # find host and path from CephFS export of form 'mon1:port,mon2:port,mon3:port:path'
+    host, export_path = share_details['export_location'].rsplit(':', 1)
```
Comment (JohnGarbutt) [inline library/os_manila_share.py:35]: ... I have a feeling the format of this radically changes at one of the ceph release boundaries. But I think this method is still safe as the path is at the end, preceded by a colon still.

File: library/os_manila_share.py [hunk 4]
```diff
@@ -1,122 +1,65 @@
...
+        rule = rules[0]
+
+    # Put required parameters at top level in result to make loops easy:
+    result['host'] = host
+    result['export'] = export_path
+    result['access_key'] = rule['access_key']
+    result['share_user'] = rule['access_to'] # supports case where user is autodetected
+    result['share_name'] = share_name
+    
+    # TODO: add more info here?
```
Comment (JohnGarbutt) [inline library/os_manila_share.py:61]: NIT: lets remove this. I would be tempted to add the read only vs read/write info from the access rule.

File: templates/ceph.conf.j2 [hunk 5]
```diff
@@ -1,5 +1,6 @@
 # {{ ansible_managed }}
 
-[client]
-	client quota = true
-	mon host = {{ os_manila_mount_host }}
+{% for share in os_manila_mount_share_info %}
+[client.{{ share.share_user }}]
```
Comment (JohnGarbutt) [inline templates/ceph.conf.j2:4]: If we have two shares that share the same share user (which is possible within a single project), will this break?

File: .travis.yml (hunk 1 above)
Comment (JohnGarbutt) [reply .travis.yml:36]: To be clear, I really mean the linting, I think that would be enough without that syntax check thingy.

**Review by JohnGarbutt — CHANGES_REQUESTED** [review]
Mostly small questions, and a request to add some CI into this... I think we have GitHub actions we can borrow from our other ansible role repos?

**Comment by JohnGarbutt** [comment]
Sorry, I meant to say... Overall this looks great, and a good step forward.

File: .travis.yml (hunk 1 above)
Comment (sjpb) [reply .travis.yml:36]: Oh I missed this. Can I add it back in later!

**Comment by sjpb** [comment]
Tested at dac9087 in a cluster, works.

**Review by JohnGarbutt — APPROVED** [review]
This looks good now, thank you.

**Pull request merged**